gpkit/fmincon/fmincon.py

```diff
@@ -0,0 +1,118 @@
+"A module to facilitate testing GPkit against fmincon"
+from gpkit import SignomialsEnabled
+from simpleflight import simpleflight
+
+def fmincon(m):
+    """A method for preparing fmincon input files to run a GPkit program"""
+
+    # Create a new dictionary mapping variables to x(i)'s for use w/ fmincon
+    i = 1
+    newdict = {}
+    newlist = []
+    original_varkeys = m.varkeys
+    for key in m.varkeys:
+        if key not in m.substitutions:
+            newdict[key] = 'x({0})'.format(i)
+            newlist += ['x_{0}: '.format(i) + key.str_without()]
+            i += 1
+
+    cost = m.cost # needs to be before subinplace()
+    constraints = m
+    constraints.subinplace(constraints.substitutions)
+    constraints.subinplace(newdict)
+
+    # Make all constraints less than zero, return list of clean strings
+    c = [] # inequality constraints
+    ceq = [] # equality constraints
+    DC = [] # gradients of inequality constraints
+    DCeq = [] # gradients of equality constraints
+    with SignomialsEnabled():
+        for constraint in constraints:
```

Comment:
As written this will not work for vector constraints; instead you might iterate over `constraints.flat(constraintsets=False)` to ensure that only (and all) inequalities are asked for their `.oper`.